### PR TITLE
feat: nft gallery blockchain filters

### PR DESCRIPTION
**Pull request by mderrick**
### 📝 Description

Add Blockchain filters to the NFT gallery.

### ❓ Context

- **Impacted projects**: `live-mobile` 
- **Linked resource(s)**: `LIVE-7462 LIVE-7464` 

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

https://github.com/LedgerHQ/ledger-live/assets/1044686/d36342a4-4f8f-4b7e-9f6e-7b587875d925

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-mobile/src/components/Nft/NftGallery/NftFilterDrawer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  readonly isOpen: boolean;
+  readonly onClose: () => void;
+  readonly toggleFilter: (filter: keyof NftGalleryChainFiltersState) => void;
+  readonly filters: NftGalleryChainFiltersState;
+};
+const NftFilterDraw: FC<Props> = ({ onClose, isOpen, filters, toggleFilter }) => {
+  const { t } = useTranslation();
+  const navigation = useNavigation();
+  return (
+    <BottomDrawer isOpen={isOpen} onClose={onClose}>
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/components/Nft/NftGallery/NftFilterDrawer/index.tsx:22]: You should use `QueuedDrawer` here :)

Comment (mderrick) [reply apps/ledger-live-mobile/src/components/Nft/NftGallery/NftFilterDrawer/index.tsx:22]: Using `QueuedDrawer` provided a myriad of problems with it self un-mounting when the NftList re-rendered after filtering. It seems to have lots of side effects that made it hard to work with. Do you have any input on this? It was much easier to use `BottomDrawer` directly and skip side effects.

File: apps/ledger-live-mobile/src/components/Nft/NftGallery/NftList.hook.ts [hunk 2]
```diff
@@ -131,6 +138,25 @@ export function useNftList({ nftList }: { nftList?: ProtoNFT[] }) {
...
+
+  const closeFilterDrawer = useCallback(() => {
+    dispatch(setGalleryFilterDrawerVisible(false));
+  }, [track, dispatch, setGalleryFilterDrawerVisible]);
+
+  const toggleChainFilter = useCallback(
+    (filter: keyof NftGalleryChainFiltersState) => {
+      dispatch(setGalleryChainFilter([filter, !chainFilters[filter]]));
+    },
+    [track, chainFilters, setGalleryChainFilter],
```
[the remark this answers is not in the document]
Comment (mderrick) [reply apps/ledger-live-mobile/src/components/Nft/NftGallery/NftList.hook.ts:157]: 👍 Yeah I'm waiting on the analytics requirements to come through and then I will fix these!

**Review by mcayuelas-ledger — APPROVED** [review]

**Comment by mcayuelas-ledger** [comment]
LGTM

**Pull request merged**